backend/api/auth.py

```
@@ -67,37 +67,68 @@ async def login(
 
 @router.post(
     "/register",
-    response_model=TokenResponse,
     status_code=status.HTTP_201_CREATED,
     summary="User Registration",
-    description="Create new user account and return JWT access token"
+    description="Create new user account - returns JWT token or email confirmation message"
 )
 async def register(
     register_data: RegisterRequest,
     db: AsyncSession = Depends(get_db)
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)